docs/reference/configuration/single-sign-on/keycloak-customization-guide.md

```diff
@@ -0,0 +1,159 @@
+
+## Customization Approaches
+
+### Approach 1: Helm Chart Values
+
+**Use for**: Configuration changes that don't require custom code
+
+**Examples**:
+- Scaling replicas or adjusting resources
+- Configuring session timeouts
+- Enabling/disabling authentication methods
+- Simple branding (logo, favicon, background)
+
+**Key Point**: Changes are applied via UDS bundle overrides and don't require rebuilding images.
+
+```yaml
+packages:
+  - name: core
+    overrides:
+      keycloak:
+        keycloak:
+          values:
+            - path: resources.requests.memory
+              value: "1Gi"
+```
+
+### Approach 2: UDS Identity Config Image
+
+**Use for**: Custom code, themes, plugins, or realm modifications
+
+**Examples**:
+- Adding custom authentication providers
+- Creating custom themes with HTML/CSS changes
+- Installing third-party Keycloak plugins
+- Modifying the default realm structure
+
+**Key Point**: Requires building a custom identity-config image from the [uds-identity-config repository](https://github.com/defenseunicorns/uds-identity-config).
+
+```yaml
+packages:
+  - name: core
+    overrides:
+      keycloak:
+        keycloak:
+          values:
+            - path: configImage
+              value: ghcr.io/your-org/identity-config:1.0.0
+```
+
+### Approach 3: OpenTofu / Infrastructure as Code
+
+**Use for**: Managing Keycloak resources declaratively after deployment
+
+**Examples**:
+- Creating clients for applications
+- Managing groups and role mappings
+- Configuring identity providers
+- Automating user provisioning
+
+**Key Point**: Manages running Keycloak resources without redeployment.
+
+### Deployment Flow:
+1. **Helm values** configure Keycloak at deployment time
+2. **Identity config image** syncs custom files via init container
+3. **OpenTofu** manages resources in the running instance
+
+**Key Points**:
+- Helm values: Applied at deployment, no image rebuild needed
+- Identity config: Requires image rebuild, applied at pod startup
+- OpenTofu: Applied to running instance, no redeployment needed
+
+## Choosing the Right Approach
+
+### Choose Helm Chart Values When:
+- You need to adjust operational settings (resources, replicas, timeouts)
+- You want to enable/disable built-in authentication methods
+- You need simple branding changes (logo, favicon, background)
+- Changes are environment-specific
+- You don't want to rebuild images
+
+### Choose Identity Config Image When:
+- You need to add custom Java plugins or libraries
+- You want extensive theme customization (HTML/CSS/JS changes)
+- You need to modify the default realm structure
+- You're adding custom PKI certificates
+- You're implementing custom Keycloak SPIs
+
+### Choose OpenTofu/IaC When:
+- You want to manage Keycloak resources as code
+- You need to automate client creation for applications
```

Comment:
I would probably remove this one, since client creation should be managed by the UDS Operator in most cases.